#!/usr/bin/env python
# http://wiki.ros.org/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile
PACKAGE = "people_detection_vision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("preprocess_resize_scale",    double_t, 0, "Preprocessing scale factor",   0.4,  0.1, 1.0)
gen.add("hog_scale_step",      double_t, 0, "Scale step",     1.05,  1.05, 1.5)
gen.add("hog_hitThreshold",    double_t, 0,
        "Threshold for the distance between features and SVM classifying plane.",     2.0, 0, 10)
gen.add("hog_finalThreshold",  double_t, 0, "Final threshold",     2.0, 0, 10)
gen.add("hog_useMeanshiftGrouping", bool_t,    0, "Data skip (px)", False)
gen.add("clusterer_data_skip", int_t,    0, "Data skip (px)", 10, 0, 20)

exit(gen.generate(PACKAGE, "people_detection_vision", "HOG"))
